ipn/ipnlocal/local_test.go

```diff
@@ -2173,3 +2175,199 @@ func TestOnTailnetDefaultAutoUpdate(t *testing.T) {
 		})
 	}
 }
+
+func TestSuggestDerpExitNode(t *testing.T) {
+	tests := []struct {
+		name       string
+		lastReport netcheck.Report
+		netMap     netmap.NetworkMap
+		wantValue  tailcfg.StableNodeID
+		wantError  error
+	}{
+		{
+			name: "returns exit node ID given exit nodes and latencies",
+			lastReport: netcheck.Report{
+				RegionLatency: map[int]time.Duration{
+					1: 10 * time.Millisecond,
+					2: 20 * time.Millisecond,
+					3: 30 * time.Millisecond,
+				},
+			},
+			netMap: netmap.NetworkMap{
+				SelfNode: (&tailcfg.Node{
+					Addresses: []netip.Prefix{
+						netip.MustParsePrefix("100.64.1.1/32"),
+						netip.MustParsePrefix("fe70::1/128"),
+					},
+				}).View(),
+				DERPMap: &tailcfg.DERPMap{
+					Regions: map[int]*tailcfg.DERPRegion{
+						1: {},
+						2: {},
+						3: {},
+						4: {},
+						5: {},
+						6: {},
+						7: {},
+						8: {},
+					},
```

Comment:
This suggests to me that maybe we could avoid using the derpmap regions and instead just infer all of the region numbers from what the peers specify.

However, the later work for non-derp peers is expected to look at the derp map so if we remove that here I suspect we'll end up re-adding it later? I'll leave this up to you.